src/vma/iomux/poll_call.cpp

```diff
@@ -181,6 +181,9 @@ bool poll_call::wait(const timeval &elapsed)
 		pto = &to;
 		m_n_all_ready_fds = orig_os_api.ppoll(m_fds, m_nfds + 1, pto, m_sigmask);
 	}else {
+		if (m_fds->fd < 0) {
+			timeout = 0;
+		}
```

Comment:
fd < 0 is used to skip os poll in (setting in the poll_call::poll_call()) and it's  okay but the timeout in this case should checked.